lex/lexer_test.go

```diff
@@ -46,6 +46,25 @@ func verifyIdentity(t *testing.T, input, expects string, isIdentity bool) {
 	}
 }
 
+func TestLexerOverrideMaxDepth(t *testing.T) {
+	{
+		l := NewLexerWithOptions(WithInput("SELECT x from y;"), WithDepthLimit(0), WithDialect(SqlDialect))
```

Comment:
[q] What does a depth limit of 0 mean?

Reply:
actually depth limit of less than 0 means do not check the limit for nested expression.